app/controllers/cards_controller.rb

```diff
@@ -0,0 +1,29 @@
+class CardsController < ApplicationController
+
+  before_filter :authenticate_user!
+
+  def edit
+  end
+
+  def update
+    customer = Stripe::Customer.retrieve(current_user.stripe_customer_id) # _token?
+    card = customer.sources.create(card: stripe_token(params))
+    card.save
+    customer.default_card = card.id
+    customer.save
+  rescue Stripe::StripeError, NoMethodError => e
+    logger.error "Stripe error while updating card info: #{e.message} for #{current_user}"
+    @error = true
+  end
+
+  private
+
+  def stripe_token(params)
+    Rails.env.test? ? generate_test_token : params[:stripeToken]
+  end
+
+  def generate_test_token
+    StripeMock.create_test_helper.generate_card_token
```

Comment:
it's not good @arreche but it's necessary for the StripeRubyMock gem https://github.com/rebelidealist/stripe-ruby-mock - I was working without it for a long time using pure VCR/PuffingBilly caching, but ultimately it got really sticky, and stripe ruby mock has been smooth sailing since, so I'm calling it a necessary evil for now ...